packages/playwright-ct-core/src/tsxTransform.ts

```diff
@@ -141,6 +143,24 @@ function collectJsxComponentUsages(node: T.Node): Set<string> {
   return names;
 }
 
+function collectClassMountUsages(node: T.Node): Set<string> {
+  const names = new Set<string>();
+  traverse(node, {
+    enter: p => {
+      // Treat calls to mount and all identifiers in arguments as component usages.
+      // e.g. mount(MyComponent, { imports: [OtherComponent], providers: [Token]})
```

Comment:
Solely relying on the file extension might not be a good idea because of:

- Support for imports/providers in Angular is a must have _[(see this for more info)](https://github.com/microsoft/playwright/pull/27783#issuecomment-2105675103)_
- Native web components don't have a common file extension: _[(see this for more info)](https://github.com/sand4rt/playwright-ct-web/blob/158588c03f9520d75bc4d22bdc3a062626b0f9b6/ct-web-lit/tests/render.spec.ts#L1-L13)_
- Vue components are not always defined in a `.vue` file _[(see this for more info)](https://github.com/microsoft/playwright/pull/30269)_

Reply:
Hmm... Could you elaborate on the imports/providers bit? I followed the link, and it seems to import `AngularComponent`, presumably from the `AngularComponent.component` file?

I also think that vue and lit are a different story, so I'd rather not mix them up in this PR aiming to support angular. We can have a separate discussion with pros and cons about it in #30269.

Reply:
I don't think that we can assume any suffix or file naming convention for these additional reasons:
- some users use different suffixes/extensions to express the responsibility of the component (e.g. `Page` & `.page.ts`)
- some users might use Analog SFCs (https://analogjs.org/docs/experimental/sfc#analog-sfcs)
- the `Component` suffix might not last (Angular team had already presented the possibility of using components in template without import `template: '<MyButton/>'` (not jsx though). If this happens, suffixes will probably be dropped.
- the `providers` option can receive classes that are not suffixed either and that should be forwarded to the browser (e.g. `mount(MyCmp, {providers: [MyService]})`) 

While the following approach works relatively well, it is limited. Nowadays, most providers are provided using functions that return the provider configuration [`provideAnimations()`](https://angular.dev/api/platform-browser/animations/provideAnimations).
There are three options:
1. keep the transformer as it is but users who want to use provider function will have to move the logic to hooks or story/test container components or create provider constants in other files: Cf. https://github.com/jscutlery/devkit/blob/353aeb322fecad64ba8b3667613deca537adb743/tests/playwright-ct-angular-demo/src/recipe-search.component.pw.ts#L34
2. forward function calls used in `providers`
3. introduce a "compiler function" like `forward(MyService)` that allows users to explicitly choose what should be transformed and "forwarded" to the browser.

Reply:
> Hmm... Could you elaborate on the imports/providers bit? I followed the link, and it seems to import AngularComponent, presumably from the AngularComponent.component file?

I meant to say that imports/providers _(especially providers)_ will receive classes that don't necessarily have a file name suffix. 

What concerns do you have regarding the current change?